src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs

```diff
@@ -362,61 +373,73 @@ protected virtual void Dispose(bool disposing)
             _serverPackageCache.PersistIfDirty();
         }
 
-        private void RebuildPackageStore()
+        private async void RebuildPackageStoreAsync(CancellationToken token)
```

Comment:
If an exception occurs in this handler, the process will terminate.  It seems like we should have some recovery and/or logging here.